app/src/main/java/com/napzak/market/presentation/registration/RegistrationViewModel.kt

```diff
@@ -0,0 +1,123 @@
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import java.text.DecimalFormat
+import javax.inject.Inject
+
+@HiltViewModel
+class RegistrationViewModel @Inject constructor(
+    /* TODO: Repository 연결 */
+) : ViewModel() {
+    private val _uiState = MutableStateFlow(RegistrationUiState())
+    val uiState = _uiState.asStateFlow()
+
+    init {
+        // TODO: 더미 데이터
+        _uiState.update {
+            it.copy(
+                genreList = listOf("건담")
+            )
+        }
+    }
+
+    fun updateTradeType(newTradeType: TradeType) = _uiState.update { it.copy(tradeType = newTradeType) }
+
+    fun updatePhotoList(newImageUrlList: List<String>) = _uiState.update { it.copy(imageUrlList = it.imageUrlList + newImageUrlList) }
+
+    fun deletePhoto(photoIndex: Int)  = _uiState.update { it.copy(imageUrlList = it.imageUrlList.filterIndexed { index, _ -> index != photoIndex }) }
+
+    fun changeRepresentPhoto(newPhoto: Int) = _uiState.update {
+        val newImageUrlList = it.imageUrlList.toMutableList()
+        newImageUrlList.add(0, newImageUrlList.removeAt(newPhoto))
+        it.copy(imageUrlList = newImageUrlList)
+    }
+
+    fun updatePlainTextValue(
+        newValue: String,
+        inputType: PlainTextInputType,
+    ) {
+        when (inputType) {
+            is PlainTextInputType.Title -> if (newValue.length <= MAX_TITLE_LENGTH) _uiState.update {
+                it.copy(title = newValue)
+            }
+            is PlainTextInputType.Description -> if (newValue.length <= MAX_DESCRIPTION_LENGTH) _uiState.update {
+                it.copy(description = newValue)
+            }
+        }
+    }
+
+    fun updateNumericValue(
+        newValue: String,
+        inputType: NumeralInputType,
+    ) {
+        when (inputType) {
+            is NumeralInputType.ProductPurchasePrice -> _uiState.update {
+                it.copy(productPurchasePrice = formatPriceValue(newValue, MAX_PURCHASE_PRICE))
+            }
+            is NumeralInputType.ProductSalePrice -> _uiState.update {
+                it.copy(productSalePrice = formatPriceValue(newValue, MAX_SALE_PRICE))
+            }
+            is NumeralInputType.NormalPostFee -> _uiState.update {
+                it.copy(normalPostFee = formatPriceValue(newValue, MAX_NORMAL_POST_FEE))
+            }
+            is NumeralInputType.HalfPostFee -> _uiState.update {
+                it.copy(halfPostFee = formatPriceValue(newValue, MAX_HALF_POST_FEE))
+            }
+        }
+    }
+
+    private fun formatPriceValue(
+        input: String,
+        maxValue: Int,
+    ): String {
+        val rawValue = input.replace(",", "").takeIf {
+            it.isNotEmpty() && it.all { it.isDigit() }
+        }?.toIntOrNull() ?: 0
+
+        return if (rawValue == 0 && input.isEmpty()) "" else {
+            val limitedValue = rawValue.coerceAtMost(maxValue)
+            DecimalFormat("#,###").format(limitedValue)
+        }
+    }
```

Comment:
```suggestion
    private fun formatPriceValue(
        input: String,
        maxValue: Int,
    ): String {
        if (input.isEmpty()) {
            return ""
        }

        val rawValue = input.replace(",", "").toIntOrNull() ?: 0

        val limitedValue = rawValue.coerceAtMost(maxValue)

        return DecimalFormat("#,###").format(limitedValue)
    }
```
p2: replace랑 takeif가 중첩되어있는 형식이 가독성 측면에서 개선이 필요한 부분인 것 같아서 이렇게 제안해봅니다! 참고만해주세요~

Reply:
너무 깔끔하네요 !! 채택하겠습니다 ㅋ